homeassistant/components/efesto/climate.py

```diff
@@ -0,0 +1,213 @@
+    def precision(self):
+        """Return the precision of the system."""
+        return PRECISION_WHOLE
+
+    @property
+    def target_temperature_step(self):
+        """Return the supported step of target temperature."""
+        return PRECISION_WHOLE
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        return TEMP_CELSIUS
+
+    @property
+    def is_on(self):
+        """Return true if on."""
+        return self._on
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._current_temperature
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return self._target_temperature
+
+    def turn_off(self):
+        """Turn device off."""
+        try:
+            json = self.client.set_off()
+            _LOGGER.debug("JSON OUTPUT: %s", json)
+            if json['status'] > 0:
+                _LOGGER.error("Failed to turn off device (%s)",
+                              json['message'])
+                return
+
+        except ValueError:
+            _LOGGER.error("Failed to turn off device")
+            return
+
+    def turn_on(self):
+        """Turn device on."""
+        try:
+            json = self.client.set_on()
+            _LOGGER.debug("JSON OUTPUT: %s", json)
+            if json['status'] > 0:
+                _LOGGER.error("Failed to turn on device (%s)", json['message'])
+                return
+
+        except ValueError:
+            _LOGGER.error("Failed to turn on device")
+            return
+
+    def set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        temperature = kwargs.get(ATTR_TEMPERATURE)
+        if temperature is None:
+            return
+        res = self.client.set_temperature(round(temperature))
+        if res['status'] > 0:
+            _LOGGER.error("Failed to set temperature")
+            return
+
+    def update(self):
+        """Get the latest data."""
+        try:
+            json = self.client.get_status()
+            _LOGGER.debug("JSON OUTPUT: %s", json)
+
+            if json['status'] == 0:
+                self._device_status = json['deviceStatus']
+                if self._device_status == 0:
+                    self._on = False
+                else:
+                    self._on = True
+                    if 'idle_info' in json:
+                        self._idle_info = json['idle_info']
+                        if self._idle_info == "TURNING OFF":
+                            self._on = False
+
+                if json['airTemperature']:
+                    self._current_temperature = json['airTemperature']
+                if json['lastSetAirTemperature']:
+                    self._target_temperature = json['lastSetAirTemperature']
+                if json['deviceStatusTranslated']:
+                    self._operation_mode = json['deviceStatusTranslated']
```

Comment:
This is an on/off device but I wanted to get the current state of what it's doing into the state in HA.
Without this it's missing critical information like alarm states, "load pellets" etc.

What I can do: set the operation mode to HEAT when device is on and save the current state in a separate variable and then override the state definition to use that variable instead of operation mode. But that makes storing the operation mode useless in my opinion, in fact operation mode is totally useless when having an on/off device that can only heat, or am I wrong?

Reply:
Operation mode should be in the standard set of modes of the platform entity.

Reply:
We're not allowed to override the state definition of the base climate component and we're only allowed to use operation mode states imported from the base climate component.

We report extra state attributes in `device_state_attributes`.